src/main/java/ch/njol/skript/expressions/ExprBeaconRange.java

```diff
@@ -0,0 +1,86 @@
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.util.Math2;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.block.Beacon;
+import org.bukkit.block.Block;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Beacon Range")
+@Description({
+	"The range of a beacon's effects, in blocks."
+})
+@Example("""
+	if the beacon tier of the clicked block is 4:
+		set the beacon effect range of the clicked block to 100
+	"""
+)
+@RequiredPlugins("Paper")
+@Since("2.10")
+public class ExprBeaconRange extends SimplePropertyExpression<Block, Double> {
+
+	private static final boolean SUPPORTS_RANGE = Skript.methodExists(Beacon.class, "getEffectRange");
+
+	static {
+		if (SUPPORTS_RANGE)
+			register(ExprBeaconRange.class, Double.class, "beacon [effect] range", "blocks");
+	}
```

Comment:
```suggestion
	static {
		if (Skript.methodExists(Beacon.class, "getEffectRange"))
			register(ExprBeaconRange.class, Double.class, "beacon [effect] range", "blocks");
	}
```
Is there a reason for the field?